### ISSUE TITLE
Invalid issuer. Expected [], received <none> (JWT::InvalidIssuerError) after upgrading from 2.8.2 to 2.9.0

### ISSUE THREAD

**Issue by tpaulshippy** [jwt/ruby-jwt#618]
Issue: InvalidIssuerError is raised even though no specific issuer was specified. There was no error in 2.8.2

Sample code to illustrate the issue:

```
require "jwt"
hmac_secret = 'my$ecretK3y'
payload = { data: "test" }

# Encode
token = JWT.encode payload, hmac_secret, "HS256"

puts token

decoded_token = JWT.decode token, hmac_secret, true, {
  verify_iss: true,
  algorithm: "HS256"
}

puts decoded_token
``` 

Running in 2.8.2 --
<img width="1267" alt="image" src="https://github.com/user-attachments/assets/984e7e6f-1775-4557-be27-5b149934aa47">

Running the same code in 2.9.0 --
<img width="1318" alt="image" src="https://github.com/user-attachments/assets/f0768a46-94e2-492d-b1d5-53412f9491ab">

**Comment by tpaulshippy** [comment]
Workaround:

```
verify_iss: false
```

**Comment by tpaulshippy** [comment]
Same issue happens with `verify_aud`

**Comment by anakinj** [comment]
Thanks for reporting. Probably some regression not covered by tests. Taking a look.

**Comment by tpaulshippy** [comment]
Yeah once I realized that my use case was not really valid (verify the issuer that I'm not actually providing), I wasn't as concerned and fixed my code to provide the issuer. But I suspect I'm not the only one who may have made this mistake. I could see a future major version actually raising an error if you ask it to verify something without providing the info needed to verify it.

**Comment by anakinj** [comment]
There has been some comments that the way the verification is enabled is not that intuitive, needing to pass both `verify_iss: true` and `iss`parameters, could be enough to just pass the expected list of issuers to enable this validation.

**Comment by anakinj** [comment]
Fixed in v2.9.1
